modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java

```diff
@@ -824,60 +842,142 @@ private Type convertIndexType(IgniteSqlIndexType type) {
     /**
      * Creates a value of required type from the literal.
      */
-    private static @Nullable Object fromLiteral(ColumnType columnType, SqlLiteral literal) {
+    private static @Nullable Object fromLiteral(ColumnType columnType, SqlLiteral literal, int precision) {
         if (literal.getValue() == null) {
             return null;
         }
 
         try {
             switch (columnType) {
-                case STRING:
-                    return literal.getValueAs(String.class);
+                case PERIOD: {
+                    if (!(literal instanceof SqlCharStringLiteral) && !(literal instanceof SqlIntervalLiteral)) {
+                        throw new SqlException(STMT_VALIDATION_ERR,
```

Comment:
I think we should come up with a better error message in case of this error (for both PERIOD and DURATION).

Reply:
Why can't we apply Default expression to interval type? Because intervals are not supported?